stl/inc/chrono

```diff
@@ -2209,6 +2221,2949 @@ namespace chrono {
+            zoned_time, decltype(_Traits2::locate_zone(string_view{})), const sys_time<_Duration>&>::value,
+            int> = 0>
+        // clang-format on
+        zoned_time(string_view _Name, type_identity_t<const sys_time<_Duration>&> _Sys)
+            : zoned_time{_Traits::locate_zone(_Name), _Sys} {}
+
+        template <class _Ptr = _TimeZonePtr,
+            enable_if_t<
+                is_convertible_v<decltype(_STD declval<_Ptr&>()->to_sys(local_time<_Duration>{})), sys_time<duration>>,
+                int> = 0>
+        zoned_time(_TimeZonePtr _Tz, const local_time<_Duration>& _Local)
+            : _Zone{_STD move(_Tz)}, _Tp{_Zone->to_sys(_Local)} {}
+
+        // clang-format off
+        template <class _Traits2 = _Traits, enable_if_t<is_constructible< // TRANSITION, VSO-1303556
+            zoned_time, decltype(_Traits2::locate_zone(string_view{})), const local_time<_Duration>&>::value,
+            int> = 0>
+        // clang-format on
+        zoned_time(string_view _Name, type_identity_t<const local_time<_Duration>>& _Local)
+            : zoned_time{_Traits::locate_zone(_Name), _Local} {}
+
+        template <class _Ptr = _TimeZonePtr,
+            enable_if_t<
+                is_convertible_v<decltype(_STD declval<_Ptr&>()->to_sys(local_time<_Duration>{}, choose::earliest)),
+                    sys_time<duration>>,
+                int> = 0>
+        zoned_time(_TimeZonePtr _Tz, const local_time<_Duration>& _Local, choose _Choose)
+            : _Zone{_STD move(_Tz)}, _Tp{_Zone->to_sys(_Local, _Choose)} {}
+
+        // clang-format off
+        template <class _Traits2 = _Traits, enable_if_t<is_constructible< // TRANSITION, VSO-1303556
+            zoned_time, decltype(_Traits2::locate_zone(string_view{})), const local_time<_Duration>&, choose>::value,
+            int> = 0>
+        // clang-format on
+        zoned_time(string_view _Name, type_identity_t<const local_time<_Duration>&> _Local, choose _Choose)
+            : zoned_time{_Traits::locate_zone(_Name), _Local, _Choose} {}
+
+        template <class _Duration2, class _TimeZonePtr2,
+            enable_if_t<is_convertible_v<sys_time<_Duration2>, sys_time<_Duration>>, int> = 0>
+        zoned_time(_TimeZonePtr _Tz, const zoned_time<_Duration2, _TimeZonePtr2>& _Zt) noexcept /* strengthened */
+            : _Zone{_STD move(_Tz)}, _Tp{_Zt.get_sys_time()} {}
+
+        template <class _Duration2, class _TimeZonePtr2,
+            enable_if_t<is_convertible_v<sys_time<_Duration2>, sys_time<_Duration>>, int> = 0>
+        zoned_time(
+            _TimeZonePtr _Tz, const zoned_time<_Duration2, _TimeZonePtr2>& _Zt, choose) noexcept /* strengthened */
+            : zoned_time{_Tz, _Zt} {}
+
+        // clang-format off
+        template <class _Duration2, class _TimeZonePtr2,
+            class _Traits2 = _Traits, enable_if_t<is_constructible< // TRANSITION, VSO-1303556

```

Comment:
The horrors

Reply:
The "recursive" constraints depicted by the Standard are indeed strange. I've added a card to the Chrono Project to investigate (after we finish C++20) whether this can be simplified to be non-recursive, which would also avoid the compiler bug that we encountered. I find `zoned_traits` to be underspecified which is the main difficulty in changing this. (We'll convert this card to an issue eventually.)